### PR TITLE
Update PHP constraints to allow v8.3

### PR DESCRIPTION
**Pull request by ADmad**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | yes
| QA            | no

### Description

Refs #175

### CODE REVIEW

**Comment by ADmad** [comment]
The "QA Checks' failure seems to be due to a bug in psalm which is addressed here https://github.com/vimeo/psalm/pull/10186 but hasn't been released yet.

File: composer.lock [hunk 1]
```diff
@@ -3998,16 +3999,16 @@
         },
         {
             "name": "vimeo/psalm",
-            "version": "5.10.0",
+            "version": "5.15.0",
```
Comment (Ocramius) [inline composer.lock]: Let's not upgrade anything: just the lock file.

Merging up is hell otherwise.

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: composer.lock (hunk 1 above)
Comment (Ocramius) [reply composer.lock]: Also, let's squash in the change that reverts all these lockfile changes :)

**Review by Ocramius — APPROVED** [review]
Thanks @ADmad!

**Comment by ADmad** [comment]
I reverted the changes and then ran `compose update --lock`, not sure why it downgraded the version for `plugin-api-version`.

**Comment by ADmad** [comment]
> I reverted the changes and then ran compose update --lock, not sure why it downgraded the version for plugin-api-version.

TIL: The value for `plugin-api-version` in `composer.lock` depends on the composer version used. I had used an older `2.0.4` version of composer hence `plugin-api-version` was set to `2.0.0`.

**Pull request merged**